### ISSUE TITLE
Tests in Reporting (non-auth) being skipped.

### ISSUE THREAD

**Issue by brizental** [webcompat/webcompat.com#1695]
So, we are skipping some tests right now, but the code we test in them is still in production.

These are the tests being skipped:

```
~ chrome on any platform - Reporting (non-auth) - details param: + decoded as space: former media type reporting test - NS_ERROR_DOM_MEDIA_DEMUXER_ERR2
~ chrome on any platform - Reporting (non-auth) - details param adds info to description: former media type reporting test - NS_ERROR_DOM_MEDIA_DEMUXER_ERR
~ chrome on any platform - Reporting (non-auth) - problem_type param selects problem type: former media type reporting test
~ firefox on any platform - Reporting (non-auth) - details param: + decoded as space: former media type reporting test - NS_ERROR_DOM_MEDIA_DEMUXER_ERR2
~ firefox on any platform - Reporting (non-auth) - details param adds info to description: former media type reporting test - NS_ERROR_DOM_MEDIA_DEMUXER_ERR
~ firefox on any platform - Reporting (non-auth) - problem_type param selects problem type: former media type reporting test
```

I think if the code is not being used it should just be removed, but if it's being used in any way it should be tested.

**Comment by miketaylr** [comment]
Thinking about this, I probably shouldn't have r+'d the patch to disable these. We currently are relying on the `details` param not exploding for some recent work @cch5ng did around stylo prefs. I also think we don't want to regress `problem_type` param behavior, even if nothing is using it.

So my vote would be to re-enable the tests -- they're not even specific to `type-media`, they just happen to look like it. They're tests around GET params that we expect to do stuff with (though it's possible nobody is currently sending those params to us right now).

**Comment by cch5ng** [comment]
@brizental @miketaylr I originally marked those tests to skip and am looking at them again

1 https://github.com/webcompat/webcompat.com/blob/master/tests/functional/reporting-non-auth.js#L274-L287
I think this one is specific to the media type reporting tool but that's fine to unskip if you prefer

2 the last 2 tests I agree are not specific to the media type reporting tool
https://github.com/webcompat/webcompat.com/blob/master/tests/functional/reporting-non-auth.js#L289-L307

and https://github.com/webcompat/webcompat.com/blob/master/tests/functional/reporting-non-auth.js#L309-L332

however, I was requested in my current reviews for #1646 to move the logic for adding details param string to the text area only on form submit. so those tests are obsolete (after #1646 is approved). I couldn't really think of a good way to test that new requirement.

**Comment by cch5ng** [comment]
chatted a little with Bea and I think for the last 2 tests (comment above), I need to make updates for #1646, and so I will also remove the skip() function on the first test at that time

**Comment by miketaylr** [comment]
> I think this one is specific to the media type reporting tool but that's fine to unskip if you prefer

Kind of. It's testing a `problem_type` GET param that the type-media tool sent. But it's not the only way it could be used in the future. And since we haven't deleted the `problem_type` support from the app itself, I don't think we should disable it.

(Note, I originally r+'d the disabling of these tests, but I think we should re-enable them).

Also, I don't think we should fix this bug in any other bug -- it should be in a PR for #1695. Otherwise git history gets confusing.

**Comment by cch5ng** [comment]
@miketaylr 
> (Note, I originally r+'d the disabling of these tests, but I think we should re-enable them).

> Also, I don't think we should fix this bug in any other bug -- it should be in a PR for #1695. Otherwise git history gets confusing.

hi, in changes for #1646, I've added back this line to the media type test:
`this.skip("former media type reporting test");`

I only removed it in my PR to save Bea the bother. the other two tests (if reenabled here) are impacted by my changes to #1646. one test will break as a result of your change request and one test becomes obsolete (it will always pass b/c the test definition becomes irrelevant) but I will change those tests back to the original.
